rust/src/trans/lipschitz_mul/mod.rs

```diff
@@ -1,70 +1,122 @@
+use num::One;
+
 use crate::{
     core::{Domain, Function, Metric, StabilityMap, Transformation},
     dist::{AbsoluteDistance, LpDistance},
     dom::{AllDomain, VectorDomain},
     error::Fallible,
-    traits::{AlertingAbs, CheckNull, DistanceConstant, SaturatingMul},
+    traits::{
+        AlertingAbs, CheckNull, ExactIntCast, FloatBits, InfAdd, InfMul, InfPow, InfSub,
+        SaturatingMul, TotalOrd,
+    },
 };
 
-pub fn make_lipschitz_mul<D, M>(l: D::Atom) -> Fallible<Transformation<D, D, M, M>>
+use super::Float;
+
+#[cfg(feature = "ffi")]
+mod ffi;
+
+pub fn make_lipschitz_mul_float<D, M>(
```

Comment:
Should bounded be in the name somewhere now? Following the pattern of the sum constructors, `make_bounded_float_lipschitz_mul` or `make_bounded_lipschitz_float_mul`?

Reply:
In this case, the input domain is not bounded. I thought we used that naming to indicate something about the input domain? Technically, we could split this constructor into a lipschitz_clamp and a lipschitz_mul... But I don't see much utility in a lipschitz_clamp on its own.

Reply:
Ah yes thanks! But sounds like we want to move the position of `mul` to match the sum constructors.